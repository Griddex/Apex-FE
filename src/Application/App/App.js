import React, { Suspense, lazy } from "react";
import { Switch, Route } from "react-router-dom";
import { Loader, Segment } from "semantic-ui-react";

const LandingView = lazy(() => import("../Routes/Landing/LandingView"));
const LoginView = lazy(() => import("../Routes/Login/LoginView"));
const Layout = lazy(() => import("../Layout/Layout"));

function App() {
  //Dimmer on loader
  return (
    <Suspense
      fallback={
        <Segment dimmed="true">
          <Loader content="Loading" />
        </Segment>
      }
    >
      <Switch>
        <Route component={LandingView} path="/" exact />
        <Route component={LoginView} path="/login" exact />
        <Route component={Layout} path="/network" exact />
        {/* <ProtectedRoute
        path={"/#"}
        roles={["Officer", "Admin"]}
        component={Layout}
      /> */}
      </Switch>
    </Suspense>
  );
}

export default App;
