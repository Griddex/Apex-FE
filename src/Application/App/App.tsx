import Button from "@material-ui/core/Button";
import { SnackbarProvider } from "notistack";
import React, { Suspense } from "react";
import { DndProvider } from "react-dnd";
import { HTML5Backend } from "react-dnd-html5-backend";
import { Route, Switch } from "react-router-dom";
import PerpetualSpinner from "./../Components/Visuals/PerpetualSpinner";

const LandingRoute = React.lazy(() => import("../Routes/Landing/LandingRoute"));
const LoginRoute = React.lazy(() => import("../Routes/Login/LoginRoute"));
const Layout = React.lazy(() => import("../Layout/Layout"));

// const useStyles = makeStyles((theme: Theme) => ({
//   success: { backgroundColor: theme.palette.tertiary.main },
//   error: { backgroundColor: theme.palette.secondary.main },
//   warning: { backgroundColor: theme.palette.quaternary.main },
//   info: { backgroundColor: theme.palette.primary.main },
// }));

const App: React.FC<JSX.Element> = () => {
  // const classes = useStyles();
  const notistackRef = React.useRef<SnackbarProvider>(null);

  return (
    <SnackbarProvider
      maxSnack={3}
      ref={notistackRef}
      action={(key) => (
        <Button
          style={{ color: "white" }}
          onClick={() => {
            if (notistackRef && notistackRef.current)
              notistackRef.current.closeSnackbar(key);
          }}
        >
          Dismiss
        </Button>
      )}
      // classes={{
      //   variantSuccess: classes.success,
      //   variantError: classes.error,
      //   variantWarning: classes.warning,
      //   variantInfo: classes.info,
      // }}
    >
      <DndProvider backend={HTML5Backend}>
        <Suspense fallback={<PerpetualSpinner message={"Loading..."} />}>
          <Switch>
            <Route exact path="/" component={LandingRoute} />
            <Route path="/login" component={LoginRoute} />
            <Route path="/apex" component={Layout} />
            {/* Not authorized will be inside protected route */}
            {/* <ProtectedRoute
        path={"/#"}
        roles={["Officer", "Admin"]}
        component={Layout}
      /> */}
            <Route render={() => <h1>Not Found</h1>} />
          </Switch>
        </Suspense>
      </DndProvider>
    </SnackbarProvider>
  );
};

export default App;
