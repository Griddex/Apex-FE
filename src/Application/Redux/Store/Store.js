import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import rootReducer from "./../Reducers/RootReducer";
import createSagaMiddleware from "redux-saga";
import watchLoginSaga from "../Sagas/LoginSaga";
import watchNewProjectSaga from "../../../Project/Redux/Sagas/NewProjectSaga";
import watchRegisterSaga from "../Sagas/RegisterSaga";
import watchConnectDatabaseSaga from "../../../Import/Redux/Sagas/connectDatabaseSaga";
import { spawn } from "redux-saga/effects";
import authMiddleware from "./../Middlewares/AuthMiddleware";
import uiSpinnerMiddleware from "./../Middlewares/UISpinnerMiddleware";
import watchRunForecastSaga from "../../../Network/Redux/Sagas/RunForecastSaga";

function* rootSaga() {
  yield spawn(watchLoginSaga);
  yield spawn(watchRegisterSaga);
  yield spawn(watchConnectDatabaseSaga);
  yield spawn(watchRunForecastSaga);
  yield spawn(watchNewProjectSaga);
}

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(uiSpinnerMiddleware, authMiddleware, sagaMiddleware)
  )
);

sagaMiddleware.run(rootSaga);

export default store;
