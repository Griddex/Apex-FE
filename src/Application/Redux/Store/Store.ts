import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import allReducers from "../Reducers/AllReducers";
import createSagaMiddleware from "redux-saga";
import watchLoginSaga from "../Sagas/LoginSaga";
import watchCreateNewProjectSaga from "../../../Project/Redux/Sagas/CreateNewProjectSaga";
import watchRegisterSaga from "../../../Administration/Redux/Sagas/RegisterSaga";
import watchConnectDatabaseSaga from "../../../Import/Redux/Sagas/ConnectDatabaseSaga";
import { spawn } from "redux-saga/effects";
import authMiddleware from "../Middlewares/AuthMiddleware";
import uiSpinnerMiddleware from "../Middlewares/UISpinnerMiddleware";
import watchRunForecastSaga from "../../../Network/Redux/Sagas/RunForecastSaga";
import rootReducer from "../Reducers/RootReducer";
import watchFetchRecentProjectsSaga from "../../../Project/Redux/Sagas/FetchRecentProjectsSaga";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";
import watchOpenRecentProjectSaga from "../../../Project/Redux/Sagas/OpenRecentProjectSaga";
import watchFetchExistingDataSaga from "../../../Import/Redux/Sagas/FetchExistingDataSaga";
import watchSaveInputDeckSaga from "../../../Import/Redux/Sagas/SaveInputDeckSaga";
import watchAutogenerateNetworkSaga from "../../../Network/Redux/Sagas/AutogenerateNetworkSaga";
import watchSaveForecastParametersSaga from "../../../Network/Redux/Sagas/SaveForecastParametersSaga";
import watchSaveNetworkSaga from "../../../Network/Redux/Sagas/SaveNetworkSaga";
import watchFetchUnitSettingsSaga from "../../../Settings/Redux/Sagas/Sagas/UnitSettingsSaga";
import watchFetchApplicationHeadersSaga from "../../../Import/Redux/Sagas/FetchApplicationHeadersSaga";
import watchAndSaveAutogenerateNetworkSaga from "../../../Network/Redux/Sagas/SaveAndAutogenerateNetworkSaga";
import watchFetchExistingForecastParametersSaga from "../../../Network/Redux/Sagas/FetchExistingForecastParametersSaga";
import watchFetchExistingNetworkDataSaga from "../../../Network/Redux/Sagas/FetchExistingNetworkDataSaga";
import watchGenerateNetworkBySelectionSaga from "../../../Network/Redux/Sagas/GenerateNetworkBySelectionSaga";
import watchUpdateForecastParametersSaga from "../../../Network/Redux/Sagas/UpdateForecastParametersSaga";
import watchSaveForecastSaga from "../../../Network/Redux/Sagas/SaveForecastSaga";
import watchGetForecastResultsSaga from "../../../Forecast/Redux/Sagas/GetForecastResultsSaga";
import watchFetchUserDetailsSaga from "../Sagas/FetchUserDetailsSaga";
import watchFetchExistingForecastingResultsSaga from "../../../Forecast/Redux/Sagas/FetchExistingForecastingResultsSaga";
import watchFetchTreeviewKeysSaga from "../../../Forecast/Redux/Sagas/FetchTreeviewKeysSaga";

function* rootSaga() {
  yield spawn(watchLoginSaga);
  yield spawn(watchRegisterSaga);
  yield spawn(watchConnectDatabaseSaga);
  yield spawn(watchCreateNewProjectSaga);
  yield spawn(watchFetchRecentProjectsSaga);
  yield spawn(watchFetchUnitSettingsSaga);
  yield spawn(watchOpenRecentProjectSaga);
  yield spawn(watchFetchExistingDataSaga);
  yield spawn(watchSaveInputDeckSaga);
  yield spawn(watchAutogenerateNetworkSaga);
  yield spawn(watchAndSaveAutogenerateNetworkSaga);
  yield spawn(watchRunForecastSaga);
  yield spawn(watchSaveForecastParametersSaga);
  yield spawn(watchUpdateForecastParametersSaga);
  yield spawn(watchSaveNetworkSaga);
  yield spawn(watchFetchApplicationHeadersSaga);
  yield spawn(watchFetchExistingNetworkDataSaga);
  yield spawn(watchFetchExistingForecastParametersSaga);
  yield spawn(watchGenerateNetworkBySelectionSaga);
  yield spawn(watchSaveForecastSaga);
  yield spawn(watchGetForecastResultsSaga);
  yield spawn(watchFetchUserDetailsSaga);
  yield spawn(watchFetchExistingForecastingResultsSaga);
  yield spawn(watchFetchTreeviewKeysSaga);
}

// function listenMongoDb() {
//   try {
//     mongoose.connect(
//       "mongodb+srv://apex:Yv10NPk7c2WmgKt4@syncware-cluster.a69fi.mongodb.net/apex-db?retryWrites=true&w=majority"
//     );

//     const db = mongoose.connection;
//     const taskCollection = db.collection("datastreams");
//     const changeStream = taskCollection.watch();
//     changeStream.on("change", (change) => {
//       console.log("Changes made to DataStream");
//     });
//   } catch (err) {
//     console.log(err);
//   }
// }

function listenWebsocket() {
  // const socket = new WebSocket()
}

const sagaMiddleware = createSagaMiddleware();

const persistConfig = {
  // configuration object for redux-persist
  key: "root",
  storage, // define which storage to use
};
// const persistedRootReducer = persistReducer(persistConfig, rootReducer);

const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(uiSpinnerMiddleware, authMiddleware, sagaMiddleware)
  )
);

// const persistor = persistStore(store);

sagaMiddleware.run(rootSaga);

// export { store, persistor };
// listenMongoDb();
export { store };
