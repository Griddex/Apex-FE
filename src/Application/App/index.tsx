import App from "./App";
import "@fontsource/open-sans";

export default App;
