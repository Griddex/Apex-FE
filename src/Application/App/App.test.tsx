import React from "react";
import { render, waitFor, cleanup } from "@testing-library/react";
import App from "./App";
import { Router } from "react-router-dom";
import history from "./../Services/HistoryService";
import { CssBaseline } from "@material-ui/core";
import { MuiPickersUtilsProvider } from "@material-ui/pickers";
import { ThemeProvider } from "@material-ui/core";
import theme from "../Theme/Theme";
import { Provider } from "react-redux";
import { store } from "../Redux/Store/Store";
import DateFnsUtils from "@date-io/date-fns";

afterEach(cleanup);

describe("Tests the root component <App/>", () => {
  const RootApp = (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <MuiPickersUtilsProvider utils={DateFnsUtils}>
          <Router history={history}>
            <CssBaseline />
            <App />
          </Router>
        </MuiPickersUtilsProvider>
      </ThemeProvider>
    </Provider>
  );

  test("renders <App/>", async () => {
    const { getByText } = render(RootApp);
    const mottoElement = await waitFor(() =>
      getByText(/...inspired technologies for business growth/i)
    );
    expect(mottoElement).toBeInTheDocument();
  });

  test("<App/> matches snapshot", () => {
    const { container } = render(RootApp);
    expect(container).toMatchSnapshot();
  });

  test("Syncware logo renders", async () => {
    const { findByAltText } = render(RootApp);
    const logoElement = await findByAltText("Syncware Logo");
    expect(logoElement).toBeInTheDocument();
  });

  test("Apex logo renders", async () => {
    const { findByAltText } = render(RootApp);
    const logoElement = await findByAltText("Apex Logo");
    expect(logoElement).toBeInTheDocument();
  });
});
