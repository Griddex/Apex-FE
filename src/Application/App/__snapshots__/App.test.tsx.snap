// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the root component <App/> <App/> matches snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-20"
  >
    <div
      class="makeStyles-header-21"
    >
      Header
    </div>
    <div
      class="makeStyles-company-22"
    >
      <img
        alt="Syncware Logo"
        height="50"
        src="SyncwareLogoWithName.svg"
      />
      <h6
        class="MuiTypography-root makeStyles-typography-25 MuiTypography-h6"
      >
        ...inspired technologies for business growth
      </h6>
    </div>
    <div
      class="makeStyles-apex-23"
    >
      <img
        alt="Apex Logo"
        height="150"
        src="ApexLogoFull.svg"
      />
      <h6
        class="MuiTypography-root makeStyles-typography-25 MuiTypography-h6"
      >
        -Syncware's flagship Hydrocarbon Forecasting and Economics Platform
      </h6>
    </div>
    <div
      class="makeStyles-footer-28"
    >
      <hr />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-27 MuiButton-disableElevation"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Proceed
        </span>
      </button>
    </div>
  </div>
</div>
`;
